### PR TITLE
A couple of fixes before everyone uses the examples

### PR DESCRIPTION
**Pull request by LRstudentHU**

### CODE REVIEW

File: code/headers/base_comm.hpp [hunk 1]
```diff
@@ -116,29 +116,54 @@ namespace r2d2 {
...
+            typename = std::enable_if_t<
+                is_suitable_frame_v<T>
+            >
+        >
+        void send(const external_id_s &id, const T &data, const priority prio = priority::NORMAL) {
+            // No {} needed, since all fields are filled.
+            // Adding it will cause a call to memset
+            frame_external_s frame;
+
+            for(uint_fast8_t i = 0; i < sizeof(T); i++){
```
Comment (itzandroidtab) [inline code/headers/base_comm.hpp:151]: is a uint_fast8_t big enough for these loops?

Comment (LRstudentHU) [reply code/headers/base_comm.hpp:151]: Hmm good point. `sizeof(T)` for for a sequence has a max of 256

File: code/headers/can.hpp [hunk 2]
```diff
@@ -343,7 +343,7 @@ namespace r2d2::can_bus {
                 frame.mode = (id >> 25) & 0x01;
                 frame.frame_type = (id >> 10) & 0xFF;
                 frame.sequence_id = (id >> 5) & 0x1F;
-                frame.sequence_total = (id >> 5) & 0x1F;
+                frame.sequence_total = id & 0x1F;
             }
                 // Standard ID, this is an error.
```
Comment (itzandroidtab) [inline code/headers/can.hpp]: this comment is probably indented to much/with tabs

File: code/headers/channel.hpp [hunk 3]
```diff
@@ -98,6 +99,38 @@ namespace r2d2::can_bus {
         // send interrupt.
         inline static queue_c<detail::_can_frame_s, 32> tx_queue;
 
+        /**
+         * Volatile flag that is used to signify that there is space
+         * in the tx_queue. The tx_queue is emptied in an interrupt and the
+         * send_frame function will wait until there is space in the tx_queue before
+         * it continues execution.
+         *
+         * However; using while(tx_queue.full()) {} will cause the compiler to
```
Comment (itzandroidtab) [inline code/headers/channel.hpp]: Is this ";" correct here?

File: code/headers/channel.hpp [hunk 4]
```diff
@@ -148,22 +179,59 @@ namespace r2d2::can_bus {
...
-            );
+                frame.length = length;
+                frame.frame_type = type;
 
-            frame.length = sizeof(T);
-            frame.frame_type = frame_type_v<T>;
+                safely_push_frame(frame);
+            } else {
+                const uint_fast8_t total = length / 8;
+                const uint_fast8_t remainder = length % 8;
```
Comment (itzandroidtab) [inline code/headers/channel.hpp:201]: is a remainder check using & 0x7 not quicker than a modulo over here?

Comment (itzandroidtab) [reply code/headers/channel.hpp:201]: apparently gcc does this automatically: [link](https://stackoverflow.com/questions/22446425/do-c-c-compilers-such-as-gcc-generally-optimize-modulo-by-a-constant-power-of)

File: code/headers/comm.hpp [hunk 5]
```diff
@@ -51,25 +51,34 @@ namespace r2d2 {
...
 
             frame_s frame{};
             frame.type = type;
 
-            memcpy(
-                (void *) frame.bytes,
-                (const void *) &data,
-                8
-            );
+            // memcpy(
```
Comment (itzandroidtab) [inline code/headers/comm.hpp]: unnecessary comment

File: code/headers/channel.hpp [hunk 6]
```diff
@@ -212,11 +281,9 @@ namespace r2d2::can_bus {
             frame.type = static_cast<frame_type>(can_frame.frame_type);
             frame.request = can_frame.mode == detail::_can_frame_mode::READ;
 
-            memcpy(
-                (void *) frame.bytes,
-                (const void *) can_frame.data.bytes,
-                8 // Has to be 8 bytes; frame.length is copied in a lower layer
-            );
+            for(uint_fast8_t i = 0; i < 8; i++){
```
Comment (itzandroidtab) [inline code/headers/channel.hpp:285]: maybe add the comment from the memcpy to the for loop

Comment (LRstudentHU) [reply code/headers/channel.hpp:285]: Solved differently, the frame is initialized to zero so we only need to copy the can_frame.length

**Review by itzandroidtab — APPROVED** [review]

**Pull request merged**